analyzer/runtime/extract.go

```diff
@@ -375,6 +376,18 @@ func ExtractRound(blockHeader nodeapi.RuntimeBlockHeader, txrs []nodeapi.Runtime
 							return fmt.Errorf("created contract: %w", err)
 						}
 					}
+					var evmEncrypted *EVMEncryptedData
+					evmEncrypted, err = EVMMaybeUnmarshalEncryptedData(body.InitCode, ok)
+					if err == nil {
```

Comment:
Nit:

```suggestion
					if evmEncrypted, err2 := EVMMaybeUnmarshalEncryptedData(body.InitCode, ok); err2 == nil {
```

I find it more readable/idiomatic, but your call. 
If you go with this, you need to also use `err2` in the `logger.Error()` below.

Same in the `EVMCall` branch.

Reply:
we need to write down some guidelines for this, because there's a lot of shadowing going on in this file and I don't get it

but I'm okay with using err2

Reply:
what about `to, err = registerRelatedEthAddress` above?

Reply:
Hehe I think that's all predominantly your code :P ... From the initial hacky import from the nexus repo.

Yeah I'm also surprised by how much shadowing we have here (I think we avoid it more, in general, hence the original comment), and also surprised that the linter doesn't complain about this instance of shadowing. I know that it sometimes does. Anyway. I'm very OK with sprucing up whatever catches your eye. I didn't want to suggest more stylistic nits because they're squarely outside the PR scope.

Reply:
rules in nexus were:
- start with `err`
- never shadow, even in cases where the linter would allow shadowing
- clobbering is okay, even through lambda boundary (nexus had not been using goroutines)
- prefer to clobber except when it interferes with a place where `:=` would allow us to omit specifying out an explicit type with `var`
- after `err` is `err1`, `err2`, etc
- use the lowest one that doesn't violate the above rules